oeps/oep-0045-arch-ops-and-config.rst

```diff
@@ -0,0 +1,169 @@
+##########################################
+OEP-45: Configuring and Operating Open edX 
+##########################################
+
+.. list-table::
+
+   * - OEP
+     - :doc:`OEP-0045 </oeps/oep-0045>`
+   * - Title
+     - Configuring and Operating Open edX
+   * - Last Modified
+     - 2020-04-06
+   * - Authors
+     - Bill DeRusha
+   * - Arbiter
+     - Felipe Montoya 
+   * - Status
+     - Under Review
+   * - Type
+     - Architecture
+   * - Created
+     - 2020-04-06
+   * - Review Period
+     - 2020-04-17 - 2020-05-04
+
+.. contents::
+   :local:
+   :depth: 3
+
+
+Abstract
+========
+
+Wherever possible, The edX organization will provide and manage Docker images as the medium for packaging operating dependencies of Open edX applications rather than providing Ansible playbooks.
+
+The configuration for running applications in different environments will be done through a config file per unique [application, environment] combination.  How those files are generated and managed is the responsibility of each operator.
+
+All standard operations for running and managing any Open edX application will be documented.  How those procedures are executed is the responsibility of each operator.
+
+Wherever possible, operating concerns will live within the codebase they are concerned with.  Thus it should be expected that all applications will provide the following:
+
+* A Dockerfile from which the container image can be built.
+* A documented settings file with production-ready defaults.
+* A documented settings file with all of the configuration values that must be defined by the operator.
+* An operations manual documenting standard operations required to run and maintain the application.
```

Comment:
Would this include / should this include a rolling changelog for changes that will affect operators?

Reply:
I think a changelog is a good idea.  I don't know the value of an operations specific log VS a single log to capture all relevant changes to the codebase.  Interested to hear other's thoughts here.

Reply:
Changelogs are often categorized. Perhaps "Operations" could be a category along with "Breaking changes", "Bug fixes", etc.

Reply:
The "django way" is to have both a changelog (release notes) and a manual that indicates differences with the previous version. 

For instance, in the manual:

> Run `foobarcocktail --no-dry-run` to update the foogarizer. This feature was introduced in release Zebulon

And then in the changelog:

> # Zebulon release notes:
> 
> [Operations][Breaking change] In order to update the foogarizer the `--no-dry-run` option must be added to the `foobarcocktail` command.
>